<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <node name="$(anon vl53l1x_scanner_driver)" pkg="vl53l1x_scanner" type="vl53l1x_scanner_driver_node"  output="screen">

        <!-- ===== [HARDWARE PARAMETERS] ===== -->
        <param name="scanner_serial_port" value="/dev/ttyACM0"/>
        <param name="scanner_serial_baud_rate" value="9600"/>
        <param name="stepper_horizontal_angle_per_step" value="0.003067962"/>
        <param name="laser_range_min" value="0.0"/>
        <param name="laser_range_max" value="4.0"/>

        <!-- ===== [ROS PARAMETERS] ===== -->
        <param name="laserscan_frame" value="laser"/>

        <!-- ===== [SCANNER PARAMETERS (WILL BE SENT TO THE SCANNER)] ===== -->
        <param name="stepper_pin_1" value="9"/>
        <param name="stepper_pin_2" value="10"/>
        <param name="stepper_pin_3" value="11"/>
        <param name="stepper_pin_4" value="12"/>
        <param name="stepper_delay" value="2250"/>
        <param name="stepper_step_min" value="-512"/>
        <param name="stepper_step_max" value="+512"/>
        <param name="laser_roi_topleftx" value="6"/>
        <param name="laser_roi_toplefty" value="15"/> <!-- TODO:long scan: 15 -->
        <param name="laser_roi_botrightx" value="9"/>
        <param name="laser_roi_botrighty" value="0"/> <!-- TODO: long scan: 0 -->
        <param name="laser_distance_mode" value="3"/>
        <param name="laser_measurement_timing_budget_micro_seconds" value="400000"/>
        <param name="laser_inter_measurement_period_milli_seconds" value="40"/>
        <param name="scanner_3d_mode" value="0"/>
        <param name="scanner_horizontal_steps_per_scan" value="16"/>
        <param name="scanner_rewind" value="1"/>
        <param name="scanner_calibration_max_value" value="500"/>

    </node>
</launch>